[V6] Add ERC20 incentive type, improve multiplier

brkagithub opened the pull request.

brkagithub committed:
Make http service use v0 routes by default

brkagithub committed:
Add ERC20 incentive type, improve multiplier

u-hubar commented on dkg/paranet.py:
```diff
@@ -259,8 +265,13 @@ def is_voter(
         self,
         ual: UAL,
         address: Address | None = None,
-        incentives_type: ParanetIncentivizationType = ParanetIncentivizationType.NEUROWEB,
```
This is a breaking change, we shouldn't do that imo

u-hubar replied:
I c your logic, but I would rather make default None and if it's None, then figure it out from the ual

u-hubar commented on dkg/paranet.py:
```diff
@@ -457,11 +508,9 @@ def update_claimable_rewards(self, ual: UAL) -> dict[str, str | HexStr | TxRecei
+        self, ual: UAL, incentives_type: ParanetIncentivizationType
     ) -> str | dict[str, str]:
-        incentives_pool_name = f"Paranet{str(incentives_type)}IncentivesPool"
+        incentives_pool_name = f"ParanetNeurowebIncentivesPool"
```
Let's move it to constants perhaps

u-hubar commented on dkg/dataclasses.py:
```diff
@@ -55,8 +55,9 @@ class KnowledgeAssetContentVisibility(AutoStrEnumUpperCase):
-    NEUROWEB = auto()
+class ParanetIncentivizationType(Enum):
+    NEUROWEB = "Neuroweb"
+    ERC20 = "NeurowebERC20"
```
Should we call it `NEUROWEB_ERC20`? There is no support for any custom ERC20 atm, so perhaps it would be more clear

br41nlet reviewed (changes requested):
Some minor adjustments needed. Also don't forget to bump the package version in the config

brkagithub committed:
Bring back incentive_type param, minor feedback

u-hubar commented on dkg/paranet.py:
```diff
@@ -91,7 +98,7 @@ def deploy_incentives_contract(
         self,
         ual: UAL,
         incentives_pool_parameters: NeuroWebIncentivesPoolParams,
-        incentives_type: ParanetIncentivizationType = ParanetIncentivizationType.NEUROWEB,
```
Let's not remove default

brkagithub committed:
bump version

brkagithub committed:
[V6] Make http service use v0 routes by default (#41)

Made API version to be '/v0' by default. Other options are '/v1' or
'/latest', also '' is option which default to v0 for backwards
compatibility reasons I believe. Tested on local network and client
functions as expected.

V8 branch will pass 'v1' for the version, or we can make it
programatically in the code decide whether to pass v0 or v1 somehow
depending on how we want to roll out v8

brkagithub committed:
Fix incentives type param to be optionally none
